lib/plugins/aws/customResources/index.js

```diff
@@ -47,9 +47,18 @@ function addCustomResourceToService(resourceName, iamRoleStatements) {
     FunctionName = `${funcPrefix}-${this.provider.naming.getCustomResourceEventBridgeHandlerFunctionName()}`;
     Handler = 'eventBridge/handler.handler';
     customResourceFunctionLogicalId = this.provider.naming.getCustomResourceEventBridgeHandlerFunctionLogicalId();
+  } else if (resourceName === 'apiGatewayCloudWatchRole') {
+    FunctionName = `${funcPrefix}-${this.provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionName()}`;
+    Handler = 'apiGatewayCloudWatchRole/handler.handler';
+    customResourceFunctionLogicalId = this.provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionLogicalId();
   } else {
     return BbPromise.reject(`No implementation found for Custom Resource "${resourceName}"`);
   }
+  if (FunctionName.length > 64) {
+    return BbPromise.reject(
+      new Error(`Resolved custom resource function name '${FunctionName}' is too long`)
```

Comment:
Would this happen if the service + stage name combo is too long? Is there anything we could do to help the user, or are they stuck?

Reply:
> Would this happen if the service + stage name combo is too long?

I believe so.

This change is purely about exposing that error earlier than it happens now (during CF deployment).

I plan to refactor a custom resources handling, and with that I'll ensure we're more reliable here, but I don't want to focus on that in context of that PR